src/components/DateInput/YearAndMonthSelector.tsx

```diff
@@ -1,30 +1,54 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import type { LocaleUtils } from 'react-day-picker';
+
+// { date, localeUtils, onChange, startYear, endYear }
+export interface YearAndMonthSelectorProps {
+  /** Selected or current date as native JS `Date` object */
+  date: Date;
+
+  /** `react-day-picker` localeUtils */
+  localeUtils: LocaleUtils;
+
+  /** Year range start as number */
+  startYear: number;
+
+  /** Year range end as number */
+  endYear: number;
+
+  /** Month/year change callback */
+  onChange?: (date: Date) => void;
+}
 
 /**
  * Private component for DateInput that renders Month/Year select inputs.
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const YearAndMonthSelector = ({ date, localeUtils, onChange, startYear, endYear }) => {
-  const months = localeUtils.getMonths();
+const YearAndMonthSelector = ({
+  date,
+  localeUtils,
+  onChange,
+  startYear,
+  endYear,
+}: YearAndMonthSelectorProps): JSX.Element => {
+  const months: ReturnType<typeof localeUtils.getMonths> = localeUtils.getMonths();
 
   const years = Array.from({ length: endYear - startYear }, (_, i) => i + startYear);
 
-  const handleChange = ({
-    target: {
-      form: { year, month },
-    },
-  }) => {
-    onChange(new Date(year.value, month.value));
+  const handleMonthChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
+    onChange?.(new Date(date.getFullYear(), Number(event.target.value)));
+  };
+
+  const handleYearChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
+    onChange?.(new Date(Number(event.target.value), date.getMonth()));
```

Comment:
Same thing here. Could assigning a noop as a default param avoid the need to optional chain here?

Reply:
yeah, same thing applies, we could probably do something like `export const noop = (...args: any[]) => {};` and import it where we need it.